src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1219 @@
+
+    # a private method to compute the exchange Laurent polynomials of `self`
+
+    def _compute_laurent(self):
+        r"""
+        Compute the exchange Laurent polynomials of ``self``.
+
+        TESTS:
+
+        This should leave the seed invariant::
+
+            sage: var('x1, x2, a')
+            (x1, x2, a)
+            sage: S = LpaSeed({x1: a, x2: a}, coefficients=[a])
+            sage: S._compute_laurent()
+            sage: S.laurent_polys()
+            [a/x2, a/x1]
+        """
+
+        # work with copies as we perform substitutions
+        exchange_polys = copy(self._exchange_polys)
+        laurent_polys = copy(self._exchange_polys)
+
+        for i in range(0, self._rank):
+
+            # this is the exchange polynomial we are
+            # finding laurent polynomial for
+            current_poly = exchange_polys[i]
+
+            # we build the laurent polynomial from scratch by iterating
+            # through the other variables, dividing by appropriate variable
+            # if necessary
+            for j in range(0, self._rank):
+                if i != j:
+
+                    # this is the polynomial we want to check
+                    # divisibility for
+                    sub_poly = current_poly.subs(
+                        **{str(self._names[j]): exchange_polys[j]})
+
+                    # calculate maximal power of exchange_polys[j] that
+                    # divides sub_poly
+                    counter = 0
+                    while True:
+                        (q, r) = sub_poly.quo_rem(
+                            exchange_polys[j] ** (counter + 1))
+                        if r != 0:
+                            break
+                        counter = counter + 1
+
+                    # divide exchange polynomial by this maximal power
+                    laurent_polys[i] = laurent_polys[i] / (
+                        self._polynomial_ring(self._names[j]) ** counter)
+
+        # after performing all substitutions, set internal laurent polynomials
+        self._laurent_polys = laurent_polys
+
+    # mutates self at ith variable
+
+    def mutate(self, i, inplace=True):
+        r"""
+        Mutate this seed at the ``i`` th index.
+
+        INPUT:
+
+        - ``i`` -- integer or iterable of integers; the index/indices to mutate
+          ``self`` at, where we index from 0.
+
+        - ``inplace`` -- boolean (default: ``True``); whether to mutate the
+          current instance of the seed, or return a new ``LpaSeed`` object.
+
+        EXAMPLES:
+
+        We mutate a rank-two Laurent phenomenon algebra at the first index::
+
+            sage: var('x1,x2,a0,a2,b0,b1')
+            (x1, x2, a0, a2, b0, b1)
+            sage: S = LpaSeed({x1: a0 + a2*x2, x2: b0 + b1*x1}, coefficients=[a0,a2,b0,b1])
+            sage: print(S)
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2*a2 + a0, x1*b1 + b0]
+            sage: S.mutate(0); print(S)
```

Comment:
```suggestion
            sage: S.mutate(0)
            sage: S
```
In general you don't need to `print` the object to get the output. (You wouldn't do that in the terminal/running Sage, right?)

Reply:
Thank you for taking the time to look through this - I really appreciate it. :)

Yes, it would be good to have some abstract seed class and implement mutations separately for cluster seeds and LP seeds, but this would involve some refactoring that I was not comfortable doing for my first PR.

I will address your suggestions ASAP.

Reply:
Be ambitious! What you are already proposing is quite a bit. Yet the natural thing would be to just refactor it;  it will need to be done anyways. (It might even have saved you some time by not duplicating stuff.)